cost/azure/rightsize_managed_disks/azure_rightsize_managed_disks.pt

```diff
@@ -0,0 +1,1081 @@
+  result "result"
+  code <<-EOS
+  var result = []
+  for (var i in ds_azure_disks_grouped_by_vm) {
+    var vmId = ds_azure_disks_grouped_by_vm[i].vmId
+    try {
+      var correspondingVm = _.find(ds_azure_virtual_machines, function(vm) { return vm.id.toLowerCase() === vmId.toLowerCase() })
+    } catch(err) {
+      continue
+    }
+    if (correspondingVm === undefined) continue;
+    var virtualMachineWithDiskData = { vmId: vmId, osDisk: null, dataDisks: [] }
+    for (j in ds_azure_disks_grouped_by_vm[i].disks) {
+      var disk = ds_azure_disks_grouped_by_vm[i].disks[j]
+      try {
+        if (correspondingVm.osDisk.managedDisk.id.toLowerCase() === disk.id.toLowerCase()) {
+          virtualMachineWithDiskData.osDisk = disk
+          continue
+        }
+      } catch (err) {
+        continue
+      }
+      try {
+        var correspondingDisk = _.find(correspondingVm.dataDisks, function(dataDisk) { return dataDisk.managedDisk.id.toLowerCase() === disk.id.toLowerCase() })
+      } catch (err) {
+        continue
+      }
+      if (correspondingDisk === undefined) continue;
+      disk.lun = correspondingDisk.lun
+      virtualMachineWithDiskData.dataDisks.push(disk)
+    }
+    result.push(virtualMachineWithDiskData)
+  }
+EOS
+end
+
+datasource "ds_azure_virtual_machines_with_metrics" do
+  iterate $ds_azure_virtual_machines_with_disks
+  request do
+    run_script $js_azure_virtual_machines_with_metrics, val(iter_item, "vmId"), $param_azure_endpoint, $param_stats_lookback
+  end
+  result do
+    encoding "json"
+    field "vmId", val(iter_item, "vmId")
+    field "metrics", jmes_path(response, "value[*]") 
+  end
+end
+
+script "js_azure_virtual_machines_with_metrics", type: "javascript" do
+  parameters "vmId", "param_azure_endpoint", "param_stats_lookback"
+  result "request"
+  code <<-EOS
+  var end_date = new Date()
+  end_date.setMilliseconds(999)
+  end_date.setSeconds(59)
+  end_date.setMinutes(59)
+  end_date.setHours(23)
+
+  var start_date = new Date()
+  start_date.setDate(start_date.getDate() - param_stats_lookback)
+  start_date.setMilliseconds(0)
+  start_date.setSeconds(0)
+  start_date.setMinutes(0)
+
+  timespan = start_date.toISOString() + "/" + end_date.toISOString()
+
+  var request = {
+    auth: "auth_azure",
+    host: param_azure_endpoint,
+    verb: "GET",
+    path: vmId + "/providers/microsoft.insights/metrics",
+    query_params: {
+      "api-version": "2018-01-01",
+      "metricnames": "OS Disk IOPS Consumed Percentage,OS Disk Bandwidth Consumed Percentage,Data Disk IOPS Consumed Percentage,Data Disk Bandwidth Consumed Percentage",
+      "timespan": timespan,
+      "$filter": "LUN eq '*'",
+      "aggregation": "Average,Maximum",
+      "interval": "FULL",
+    },
+    headers: {
+      "User-Agent": "RS Policies",
+    },
+    ignore_status: [400, 403, 404],
+  }
+EOS
+end
+
+datasource "ds_azure_virtual_machines_with_disk_space" do
```

Comment:
Why is this here? Looks like it does nothing but return an empty list

Reply:
Hey @XOmniverse, currently we want to leave this "empty datasource" that is just returning an empty list, this would be replaced by an API call to download a file containing a JSON list that would contain information about the disk remaining capacity (GiB), something like this:

![image](https://github.com/flexera-public/policy_templates/assets/54189123/380f44c6-5844-4db5-8d87-aed72ed7d189)

So since this change will be applied soon (we just need to verify a couple things with client), we would like to leave that code in there since we already tested it and make sure it works.

I would add something to the README.md telling that this feature is not yet available.